pandas/_libs/internals.pyx

```diff
@@ -370,22 +378,17 @@ def get_blkno_indexers(int64_t[:] blknos, bint group=True):
     # There's blkno in this function's name because it's used in block &
     # blockno handling.
     cdef:
-        int64_t cur_blkno
-        Py_ssize_t i, start, stop, n, diff
-
+        Py_ssize_t i, diff, tot_len, stop
+        Py_ssize_t start = 0, n = blknos.shape[0]
         object blkno
         object group_dict = defaultdict(list)
         int64_t[:] res_view
-
-    n = blknos.shape[0]
+        int64_t cur_blkno = blknos[start]
 
     if n == 0:
         return
 
-    start = 0
-    cur_blkno = blknos[start]
-
-    if group is False:
```

Comment:
`if group is False` should be marginally faster than `if not group`

Reply:
From my benchmarking it's actually the other way around:

```
In [1]: foo = True                                                                                            

In [2]: %timeit foo is False                                                                                  
22.9 ns ± 0.157 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [3]: %timeit foo is False                                                                                  
23 ns ± 0.0388 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [4]: %timeit foo is False                                                                                  
22.7 ns ± 0.194 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [5]: %timeit not foo                                                                                       
20.7 ns ± 0.0589 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [6]: %timeit not foo                                                                                       
21.1 ns ± 0.0636 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [7]: %timeit not foo                                                                                       
20.6 ns ± 0.0497 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
```

---

```

In [1]: bar = False                                                                                           

In [2]: %timeit bar is False                                                                                  
28.5 ns ± 0.077 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [3]: %timeit bar is False                                                                                  
28.6 ns ± 0.0666 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [4]: %timeit bar is False                                                                                  
29.5 ns ± 0.203 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [5]: %timeit not bar                                                                                       
26.1 ns ± 0.217 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [6]: %timeit not bar                                                                                       
25.8 ns ± 0.0231 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [7]: %timeit not bar                                                                                       
26.7 ns ± 0.0831 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
```

Reply:
look at the generated C code.  The "is" should be a pointer comparison, whereas the other one should have to do some more work.